src/ecmult_impl.h

```diff
@@ -106,13 +113,11 @@ static void secp256k1_ecmult_odd_multiples_table(int n, secp256k1_gej *prej, sec
  *  and use the precomputed table in <ecmult_static_pre_g.h> for G.
  */
 static void secp256k1_ecmult_odd_multiples_table_globalz_windowa(secp256k1_ge *pre, secp256k1_fe *globalz, const secp256k1_gej *a) {
-    secp256k1_gej prej[ECMULT_TABLE_SIZE(WINDOW_A)];
     secp256k1_fe zr[ECMULT_TABLE_SIZE(WINDOW_A)];
 
     /* Compute the odd multiples in Jacobian form. */
-    secp256k1_ecmult_odd_multiples_table(ECMULT_TABLE_SIZE(WINDOW_A), prej, zr, a);
-    /* Bring them to the same Z denominator. */
-    secp256k1_ge_globalz_set_table_gej(ECMULT_TABLE_SIZE(WINDOW_A), pre, globalz, prej, zr);
+    secp256k1_ecmult_odd_multiples_table(ECMULT_TABLE_SIZE(WINDOW_A), pre, zr, globalz, a);
+    secp256k1_ge_globalz_fixup_table(ECMULT_TABLE_SIZE(WINDOW_A), pre, zr);
 }
 
 /** The following two macro retrieves a particular odd multiple from a table
```

Comment:
Or just turn the macros into functions... I'm not aware of a real reason to use these procedural macros in modern C.*



* Unless you use macros to express things not expressible with a function, e.g., making up identifiers on the fly via string concatenation.

Reply:
I believe we got here as follows.  The macro was introduce in https://github.com/bitcoin-core/secp256k1/commit/b1483f874c6111b7d727d6c6cb3df35c16a880bc#diff-1f71a59e139573dfb0695be308c69455971378cd67ad7946a3703cb725b589fdR51 (expand the diff for ecmult.c), where there was a function name argument being passed in.  I suppose in that situation a macro makes sense to avoid the use of function pointers?

But this function argument was eliminated in https://github.com/bitcoin-core/secp256k1/commit/41f84554341c3c2f21324da53982b8da045f88e3#diff-dc206b3f5f973b97a0f5c1b12d705d0e58a1cb8c9684e481e7a34fc05e4716aaL63 at which point there is no longer a reason for this to continue to be a macro.

Thus it should be safe for me to convert it into an inline function.

Reply:
Should I move the macros into function and #define ECMULT_TABLE_VERIFY?

Reply:
That would be ideal I think, but turning the macros to functions isn't necessary for this PR. I just suggested isolating `ECMULT_TABLE_VERIFY` because it simplifies review of this PR.

Reply:
Looks like you did both, thanks.